src/backend/src/services/statistics.services.ts

```diff
@@ -115,4 +117,15 @@ export default class StatisticsService {
       };
     });
   }
+
+  static async getSingleGraph(id: string, organization: Organization): Promise<Graph> {
+    const requestedGraph = await prisma.graph.findUnique({
+      where: { id, organizationId: organization.organizationId },
+      ...getGraphQueryArgs(organization.organizationId)
+    });
+
+    if (!requestedGraph) throw new NotFoundException('Graph', id);
+
+    return graphTransformer(requestedGraph as Prisma.GraphGetPayload<GraphQueryArgs> & { graphData: GraphData[] });
```

Comment:
you shouldn’t have to use an as here. if your missing the graphData you probably need to retrieve that. which you should be able to use by using the getGraphData function above

Reply:
<img width="606" alt="Screenshot 2024-12-12 at 12 22 18 PM" src="https://github.com/user-attachments/assets/1bb3cdbe-ab5a-4612-a99f-2c1b9108d974" />
fixed the 'as', but I don't fundamentally understand the query path - for GraphGen its a single QueryPath but the Graph schema has queryPaths Graph_Query[]

Reply:
theres a comment in the schema explaining how that works, but i think were gonna get rid of that actually

Reply:
tried my best to convert, however I'm having this new issue where none of the stat tests (even the ones I didn't write) aren't passing <img width="768" alt="Screenshot 2024-12-12 at 7 48 02 PM" src="https://github.com/user-attachments/assets/ade3e9bb-29a2-4d70-887d-83a2f49dbe14" />

Reply:
tried my best to convert - not sure how to go about fixing the test that failed